richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
@afester Since my above approach didn't work, I think we will have to do something at the `ReadOnlyStyledDocument` level. We might even need to make `Paragraph#subsequence` return an `Optional<Paragraph<PS, SEG, S>>`. However, I'm not as familiar with how that code in `ReadOnlyStyledDocument` works since it uses `Tuple3`, so it's not easy to follow the logic as to what each argument is. Then again, I only glanced it and haven't really studied it.